ts/features/fci/screens/valid/FciDocumentsScreen.tsx

```diff
@@ -1,17 +1,168 @@
 import * as React from "react";
-import { SafeAreaView, ScrollView } from "react-native";
-import { H1 } from "../../../../components/core/typography/H1";
+import Pdf from "react-native-pdf";
+import { constNull, pipe } from "fp-ts/lib/function";
+import * as O from "fp-ts/lib/Option";
+import { SafeAreaView, StyleSheet } from "react-native";
+import { useSelector } from "react-redux";
+import { useNavigation } from "@react-navigation/native";
+import IconFont from "../../../../components/ui/IconFont";
 import { IOStyles } from "../../../../components/core/variables/IOStyles";
 import BaseScreenComponent from "../../../../components/screens/BaseScreenComponent";
+import { IOColors } from "../../../../components/core/variables/IOColors";
+import FooterWithButtons from "../../../../components/ui/FooterWithButtons";
+import I18n from "../../../../i18n";
+import DocumentsNavigationBar from "../../components/DocumentsNavigationBar";
+import TouchableDefaultOpacity from "../../../../components/TouchableDefaultOpacity";
+import { emptyContextualHelp } from "../../../../utils/emptyContextualHelp";
+import { fciSignatureDetailDocumentsSelector } from "../../store/reducers/fciSignatureRequest";
 
-const FciDocumentsScreen = () => (
-  <BaseScreenComponent goBack={true} headerTitle={"Documenti"}>
-    <SafeAreaView style={IOStyles.flex} testID={"FciDocumentsScreenTestID"}>
-      <ScrollView style={IOStyles.horizontalContentPadding}>
-        <H1>{"FciDocumentsScreen"}</H1>
-      </ScrollView>
-    </SafeAreaView>
-  </BaseScreenComponent>
-);
+const styles = StyleSheet.create({
+  pdf: {
+    flex: 1,
+    backgroundColor: IOColors.bluegrey
+  }
+});
 
+const FciDocumentsScreen = () => {
+  const pdfRef = React.useRef<Pdf>();
+  const [totalPages, setTotalPages] = React.useState(0);
+  const [currentPage, setCurrentPage] = React.useState(0);
+  const [currentDoc, setCurrentDoc] = React.useState(0);
+  const documents = useSelector(fciSignatureDetailDocumentsSelector);
+  const navigation = useNavigation();
+
+  // TODO: abort signature flow
+  const onCancelPress = () => constNull;
+
+  const cancelButtonProps = {
+    block: true,
+    light: false,
+    bordered: true,
+    onPress: onCancelPress,
+    title: I18n.t("global.buttons.cancel")
+  };
+
+  // TODO: navigate to signature fields selection screen
+  const onContinuePress = () => constNull;
+
+  const continueButtonProps = {
+    block: true,
+    primary: true,
+    onPress: onContinuePress,
+    title: I18n.t("global.buttons.continue")
+  };
+
+  const keepReadingButtonProps = {
+    block: true,
+    light: true,
+    bordered: true,
+    onPress: () =>
+      pipe(
+        pdfRef.current,
+        O.fromNullable,
+        O.map(_ => _.setPage(totalPages))
+      ),
+    title: I18n.t("global.buttons.continue")
+  };
+
+  const renderPager = () => (
+    <Pdf
```

Comment:
What about extracting this into its own component? (you can use [forwardRef](https://reactjs.org/docs/forwarding-refs.html) in case to share the ref with its parent).